# Nomes: Luca Wolffenbuttel Bohnenberger e Louise Zanol Northfleet

In [15]:
from statsmodels.stats.multivariate import test_mvmean_2indep
import pandas as pd 

In [16]:
df = pd.read_csv("laseri.csv")
df 

Sex  Age  Height  Weight  Waist    Hip  PWVT3  PWVT4   BMI    WHR  \
0      Male   33     181      88   86.9  102.0   17.1   10.9  26.9  0.852   
1    Female   30     152      58   76.5   91.5   13.2    7.4  25.1  0.836   
2    Female   33     174     104   85.4  114.5   13.3    8.1  34.4  0.746   
3    Female   36     168      81   90.6  107.7   16.0   16.0  28.7  0.841   
4    Female   36     172      90   88.1  107.8   15.8    9.0  30.4  0.817   
..      ...  ...     ...     ...    ...    ...    ...    ...   ...    ...   
218    Male   28     186      96   97.1  107.4   19.0   19.0  27.7  0.904   
219  Female   34     154      63   75.1   93.5   14.0   11.4  26.6  0.803   
220  Female   40     164      86   94.6  118.5   14.3    9.1  32.0  0.798   
221  Female   37     163      57   68.5   91.9   14.4    8.4  21.5  0.745   
222  Female   41     171      80   89.0  111.4   14.9   10.0  27.4  0.799   

     ...  PWVT1  PWVT2  HRT1T4  COT1T4  SVRIT1T4  PWVT1T4  HRT1T2  COT1T2  \
0    ...   10.6   19.7      11    1.35      -470     -0.3      -5    1.10   
1    ...    7.2   11.6       0    0.33      -130     -0.2       0    1.86   
2    ...    7.0   11.1       1    1.36      -407     -1.1      -6    2.23   
3    ...    9.1   16.7     -13    1.28      -587     -6.9     -14    1.64   
4    ...    9.0   15.6       4    1.07      -305      0.0      -8    2.93   
..   ...    ...    ...     ...     ...       ...      ...     ...     ...   
218  ...   10.3   16.6     -26    0.82      -488     -8.7     -18    0.36   
219  ...   10.5   14.2       2    0.32      -480     -0.9       1    0.39   
220  ...    8.8   15.0       1    0.14      -132     -0.3     -14   -0.19   
221  ...    7.9   13.1      12    1.33      -286     -0.5     -32    3.07   
222  ...    9.3   16.3      11    1.45      -477     -0.7      -4    1.81   

     SVRIT1T2  PWVT1T2  
0        -755     -9.1  
1       -1258     -4.4  
2        -882     -4.1  
3        -921     -7.6  
4       -1171     -6.6  
..        ...      ...  
218      -508     -6.3  
219      -609     -3.7  
220      -433     -6.2  
221     -1058     -5.2  
222      -692     -7.0  

[223 rows x 32 columns]

In [17]:
df_a = df[[ "Sex","HRT1", "HRT2", "HRT3", "HRT4"]]
df_a

Sex  HRT1  HRT2  HRT3  HRT4
0      Male    75    80    91    64
1    Female    65    65    82    65
2    Female    67    73    79    66
3    Female    61    75    74    74
4    Female    68    76    80    64
..      ...   ...   ...   ...   ...
218    Male    71    89    97    97
219  Female    72    71    73    70
220  Female    57    71    73    56
221  Female    73   105   103    61
222  Female    66    70    80    55

[223 rows x 5 columns]

In [18]:
df_fem = df[df["Sex"] == "Female"]
df_fem = df_fem.drop(columns="Sex")
df_fem

Age  Height  Weight  Waist    Hip  PWVT3  PWVT4   BMI    WHR  HRT1  ...  \
1     30     152      58   76.5   91.5   13.2    7.4  25.1  0.836    65  ...   
2     33     174     104   85.4  114.5   13.3    8.1  34.4  0.746    67  ...   
3     36     168      81   90.6  107.7   16.0   16.0  28.7  0.841    61  ...   
4     36     172      90   88.1  107.8   15.8    9.0  30.4  0.817    68  ...   
5     33     165     101  124.7  118.1   10.8    8.8  37.1  1.056    57  ...   
..   ...     ...     ...    ...    ...    ...    ...   ...    ...   ...  ...   
216   26     161      56   69.3   90.3   13.7    7.4  21.6  0.767    55  ...   
219   34     154      63   75.1   93.5   14.0   11.4  26.6  0.803    72  ...   
220   40     164      86   94.6  118.5   14.3    9.1  32.0  0.798    57  ...   
221   37     163      57   68.5   91.9   14.4    8.4  21.5  0.745    73  ...   
222   41     171      80   89.0  111.4   14.9   10.0  27.4  0.799    66  ...   

     PWVT1  PWVT2  HRT1T4  COT1T4  SVRIT1T4  PWVT1T4  HRT1T2  COT1T2  \
1      7.2   11.6       0    0.33      -130     -0.2       0    1.86   
2      7.0   11.1       1    1.36      -407     -1.1      -6    2.23   
3      9.1   16.7     -13    1.28      -587     -6.9     -14    1.64   
4      9.0   15.6       4    1.07      -305      0.0      -8    2.93   
5      8.3   11.4       8    0.72      -635     -0.5     -16    0.06   
..     ...    ...     ...     ...       ...      ...     ...     ...   
216    7.8   12.5       4    0.31      -238      0.4     -14    0.32   
219   10.5   14.2       2    0.32      -480     -0.9       1    0.39   
220    8.8   15.0       1    0.14      -132     -0.3     -14   -0.19   
221    7.9   13.1      12    1.33      -286     -0.5     -32    3.07   
222    9.3   16.3      11    1.45      -477     -0.7      -4    1.81   

     SVRIT1T2  PWVT1T2  
1       -1258     -4.4  
2        -882     -4.1  
3        -921     -7.6  
4       -1171     -6.6  
5        -467     -3.1  
..        ...      ...  
216      -496     -4.7  
219      -609     -3.7  
220      -433     -6.2  
221     -1058     -5.2  
222      -692     -7.0  

[120 rows x 31 columns]

In [19]:
df_mas = df[df["Sex"] == "Male"]
df_mas = df_mas.drop(columns="Sex")
df_mas

Age  Height  Weight  Waist    Hip  PWVT3  PWVT4   BMI    WHR  HRT1  ...  \
0     33     181      88   86.9  102.0   17.1   10.9  26.9  0.852    75  ...   
7     30     178      86   84.7  105.5   11.8    6.9  27.1  0.803    47  ...   
8     30     170      76   80.5   93.5   17.6    8.8  26.3  0.861    71  ...   
10    36     172      65   83.4   86.4   17.1    9.9  22.0  0.965    69  ...   
11    36     172      75   85.7   99.5   17.2   11.1  25.4  0.861    54  ...   
..   ...     ...     ...    ...    ...    ...    ...   ...    ...   ...  ...   
209   35     184     100  102.2  106.5   17.4   13.3  29.5  0.960    70  ...   
214   32     171      58   84.2   93.8   14.4    9.0  19.8  0.898    49  ...   
215   29     180      86   86.5  101.3   17.5   10.4  26.5  0.854    58  ...   
217   35     168      80  102.7  103.0   17.8   11.5  28.3  0.997    85  ...   
218   28     186      96   97.1  107.4   19.0   19.0  27.7  0.904    71  ...   

     PWVT1  PWVT2  HRT1T4  COT1T4  SVRIT1T4  PWVT1T4  HRT1T2  COT1T2  \
0     10.6   19.7      11    1.35      -470     -0.3      -5    1.10   
7      6.8   11.2       1    0.86      -299     -0.1     -10    1.42   
8      8.5   15.9       8    1.07      -228     -0.3     -10    1.85   
10    10.2   14.9       2    0.27      -141      0.3      -8    1.13   
11    11.1   15.3       3    0.57      -297      0.0     -14    0.47   
..     ...    ...     ...     ...       ...      ...     ...     ...   
209   12.9   14.3       8    0.61      -544     -0.4     -13    0.11   
214    8.6   14.1       1    0.38      -437     -0.4     -23    0.06   
215   10.1   15.8       0    0.11       114     -0.3     -13    0.08   
217   11.4   15.3       5    0.27       -54     -0.1      -4    0.07   
218   10.3   16.6     -26    0.82      -488     -8.7     -18    0.36   

     SVRIT1T2  PWVT1T2  
0        -755     -9.1  
7       -1006     -4.4  
8        -633     -7.4  
10       -609     -4.7  
11       -405     -4.2  
..        ...      ...  
209      -526     -1.4  
214       -44     -5.5  
215      -350     -5.7  
217       -29     -3.9  
218      -508     -6.3  

[103 rows x 31 columns]

## Primeira questao

#### Hotellings test for multivariate mean in two independent samples

#### The null hypothesis is that both samples have the same mean. The alternative hypothesis is that means differ.

In [20]:
#a)
df_a_mas = df_mas[["HRT1", "HRT2", "HRT3", "HRT4"]]
df_a_fem = df_fem[["HRT1", "HRT2", "HRT3", "HRT4"]]
test_mvmean_2indep(df_a_mas, df_a_fem)

<class 'statsmodels.stats.base.HolderTuple'>
statistic = np.float64(1.4892684332610164)
pvalue = np.float64(0.20643536815038813)
df = (4, 218)
t2 = np.float64(6.039051811939168)
distr = 'F'
tuple = (np.float64(1.4892684332610164), np.float64(0.20643536815038813))

In [21]:
#b)
df_b_mas = df_mas[["COT1", "COT2", "COT3", "COT4"]]
df_b_fem = df_fem[["COT1", "COT2", "COT3", "COT4"]]
test_mvmean_2indep(df_b_mas, df_b_fem)

<class 'statsmodels.stats.base.HolderTuple'>
statistic = np.float64(33.023299629871325)
pvalue = np.float64(1.5859674105444531e-21)
df = (4, 218)
t2 = np.float64(133.91099482938648)
distr = 'F'
tuple = (np.float64(33.023299629871325), np.float64(1.5859674105444531e-21))

In [22]:
# c) Rejeita H0 
df_c_mas = df_mas[["SVRIT1", "SVRIT2", "SVRIT3", "SVRIT4"]]
df_c_fem = df_fem[["SVRIT1", "SVRIT2", "SVRIT3", "SVRIT4"]]
test_mvmean_2indep(df_c_mas, df_c_fem)

<class 'statsmodels.stats.base.HolderTuple'>
statistic = np.float64(4.9333613486953345)
pvalue = np.float64(0.000794299516709327)
df = (4, 218)
t2 = np.float64(20.005006569938878)
distr = 'F'
tuple = (np.float64(4.9333613486953345), np.float64(0.000794299516709327))

In [23]:
# d) Rejeita H0 
df_d_mas = df_mas[["PWVT1", "PWVT2", "PWVT3", "PWVT4"]]
df_d_fem = df_fem[["PWVT1", "PWVT2", "PWVT3", "PWVT4"]]
test_mvmean_2indep(df_d_mas, df_d_fem)

<class 'statsmodels.stats.base.HolderTuple'>
statistic = np.float64(9.538709035172372)
pvalue = np.float64(3.9940186934097814e-07)
df = (4, 218)
t2 = np.float64(38.6799026930843)
distr = 'F'
tuple = (np.float64(9.538709035172372), np.float64(3.9940186934097814e-07))

## Questão 2

In [24]:
df = pd.read_excel("/Users/louisenorthfleet/Desktop/data_2/data.xlsx")
df

X1     X2     X3      X4   X5  X6   X7  X8     Y1     Y2
0    0.98  514.5  294.0  110.25  7.0   2  0.0   0  15.55  21.33
1    0.98  514.5  294.0  110.25  7.0   3  0.0   0  15.55  21.33
2    0.98  514.5  294.0  110.25  7.0   4  0.0   0  15.55  21.33
3    0.98  514.5  294.0  110.25  7.0   5  0.0   0  15.55  21.33
4    0.90  563.5  318.5  122.50  7.0   2  0.0   0  20.84  28.28
..    ...    ...    ...     ...  ...  ..  ...  ..    ...    ...
763  0.64  784.0  343.0  220.50  3.5   5  0.4   5  17.88  21.40
764  0.62  808.5  367.5  220.50  3.5   2  0.4   5  16.54  16.88
765  0.62  808.5  367.5  220.50  3.5   3  0.4   5  16.44  17.11
766  0.62  808.5  367.5  220.50  3.5   4  0.4   5  16.48  16.61
767  0.62  808.5  367.5  220.50  3.5   5  0.4   5  16.64  16.03

[768 rows x 10 columns]

In [32]:
import statsmodels.api as sm

variaveis = [col for col in df.columns if col not in ['Y1', 'Y2']]

X = df[variaveis]

#X.drop(["X6","X8"], axis=1, inplace=True)

X = sm.add_constant(X)

y = df['Y2']

# Ajusta OLS
fit0 = sm.OLS(y, X).fit()

print(fit0.summary())

                            OLS Regression Results                            
Dep. Variable:                     Y2   R-squared:                       0.888
Model:                            OLS   Adj. R-squared:                  0.887
Method:                 Least Squares   F-statistic:                     859.1
Date:                Mon, 17 Nov 2025   Prob (F-statistic):               0.00
Time:                        18:38:23   Log-Likelihood:                -1979.3
No. Observations:                 768   AIC:                             3975.
Df Residuals:                     760   BIC:                             4012.
Df Model:                           7                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const         97.2457     20.765      4.683      0.0

para X6 o intervalo é:

[-0.081 ; 0.324]


Para X8:

[-0.109 ; 0.190]


Se o intervalo contém zero →
pode ser positivo, negativo ou nulo, ou seja, o modelo não sabe se a variável aumenta ou diminui Y.

📌 Outro indicativo de que X6 e X8 não têm efeito confiável.